python/delta_sharing/rest_client.py

```diff
@@ -227,27 +229,43 @@ def list_all_tables(
     @retry_with_exponential_backoff
     def query_table_metadata(self, table: Table) -> QueryTableMetadataResponse:
         with self._get_internal(
-            f"/shares/{table.share}/schemas/{table.schema}/tables/{table.name}/metadata"
-        ) as lines:
+            f"/shares/{table.share}/schemas/{table.schema}/tables/{table.name}/metadata",
+            return_headers=True
+        ) as values:
+            headers = values[0]
+            # it's a bug in the server if it doesn't return delta-table-version in the header
+            if "delta-table-version" not in headers:
+                raise LookupError("Missing delta-table-version header")
```

Comment:
Question: Can you remind me why we decided to return the table version in the header?

Reply:
Hmmm, I don't know. I guess it was for light request/response?
@zsxwing Do you still remember this?